React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,498 @@
+# Chakra UI
+
+Chakra UI is a simple, modular and accessible component library that gives you the building blocks you need to build your React applications.
+
+## Getting started
+
+### Installation:
+
+```
+npm i @chakra-ui/react @emotion/react @emotion/styled framer-motion
+```
+
+### Setup provider:
+
+For Chakra UI to work correctly, you need to setup the `ChakraProvider` at the root of your application.
+
+Example:
+
+```jsx
+import { ChakraProvider } from "@chakra-ui/react";
+
+function App({ Component, pageProps }: AppProps): ReactElement {
+  return (
+    <ChakraProvider theme={theme}>
+      <Component {...pageProps} />
+    </ChakraProvider>
+  );
+}
+
+export default App;
+```
+
+Note:
+
+- For Next.js, you need to set this up in `pages/_app.tsx`
+- For Create React App, you need to set this up in `index.tsx`
+
+### Style props
+
+Style props are a way to alter the style of a component by simply passing props to it. It helps to save time by providing helpful shorthand ways to style components.
+
+Alternative to that is using sx prop that takes an object with style rules.
+
+```tsx
+// shorthand width prop
+<Box w="400px">Lorem Ipsum is simply dummy text</Box>
+
+// sx prop
+<Box sx={{ width: '400px'}}>Lorem Ipsum is simply dummy text</Box>
+```
+
+For more about style props read [docs](https://chakra-ui.com/docs/features/style-props)
+
+#### The `as` prop
+
+The `as` prop is a feature that all Chakra UI components have and it allows you to pass an HTML tag or component to be rendered.
+
+For example, say you are using a Button component, and you need to make it a link instead. You can compose a and Button like this:
```

Comment:
To make it clearer that `a` is not the article from English language (a/the), but the `anchor` element:

```suggestion
For example, say you are using a `Button` component, and you need to make it a link instead. You can compose `a` and `Button` like this:
```